service/storage_service.cc

```diff
@@ -3608,8 +3611,12 @@ future<> storage_service::check_for_endpoint_collision(std::unordered_set<gms::i
     });
 }
 
-future<> storage_service::remove_endpoint(inet_address endpoint, gms::permit_id pid) {
-    co_await _gossiper.remove_endpoint(endpoint, pid);
+future<> storage_service::remove_endpoint(inet_address endpoint, gms::permit_id pid, bool force) {
+    if (force) {
+        co_await _gossiper.force_remove_endpoint(endpoint, pid);
```

Comment:
Let's not use the force_remove_endpoint function internally.

Reply:
> It is used ONLY for the restful API. It is not supposed to be used to remove entries in gossip normally.

Why not reuse it for other purposes?